sound/soc/sof/intel/hda-codec.c

```diff
@@ -151,6 +151,8 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 		if (!hdev->bus->audio_component) {
 			dev_dbg(sdev->dev,
 				"iDisp hw present but no driver\n");
+
+			snd_hdac_ext_bus_device_exit(hdev);
```

Comment:
Should we call ``snd_hdac_ext_bus_device_exit(hdev);`` if ret != 0 before [return ret;](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda-codec.c#L180), too?

Reply:
No, it is already called in snd_hdac_ext_bus_device_init if ret < 0 for one case. We need to add it in snd_hdac_ext_bus_device_init if snd_hdac_device_init is failed. It is another PR.

Reply:
No, we can't call snd_hdac_ext_bus_device_exit if snd_hdac_device_init is failed since the snd_hdac_ext_bus_device_exit is not fit for this case.

Reply:
yes,  this is another case.